src/hotspot/share/runtime/mutex.cpp

```diff
@@ -384,14 +384,20 @@ void Mutex::check_rank(Thread* thread) {
   if (owned_by_self()) {
     // wait() case
     Mutex* least = get_least_ranked_lock_besides_this(locks_owned);
-    // We enforce not holding locks of rank nosafepoint or lower while waiting.
+    // We enforce not holding locks of rank nosafepoint or lower while waiting for JavaThreads,
+    // because the held lock has a NoSafepointVerifier so waiting on a lock will block out safepoints.
+    // For NonJavaThreads, we enforce not waiting on the tty lock since that could block progress also.
```

Comment:
You are checking this for all threads not just NJTs. Is the comment wrong or the code?

Reply:
I could start the comment like, For JavaThreads, we enforce not holding ... and leave the For NonJavaThreads ...